scheduler/server/orig_scheduling_alg.go

```diff
@@ -7,10 +7,47 @@ import (
 	"github.com/twitter/scoot/scheduler/domain"
 )
 
-type OrigSchedulingAlg struct{}
+// Decrease the NodeScaleFactor by taking the percentage defined by NodeScaleAdjust[Priority].
+// Note: the use case here is to hit an SLA for each job priority, and this is a coarse way to do so.
+var DefaultNodeScaleAdjustment = []float32{.15, .3, .55}
 
+// OrigSchedulingAlgConfig configuration for the original scheduling algorithm
+type OrigSchedulingAlgConfig struct {
+	SoftMaxSchedulableTasks int
+	NodeScaleAdjustment     []float32
+}
+
+// OrigSchedulingAlg the original scheduling algorithm
+type OrigSchedulingAlg struct {
+	Config *OrigSchedulingAlgConfig
+}
+
+// GetTasksToBeAssigned the the tasks to be started
 func (oa *OrigSchedulingAlg) GetTasksToBeAssigned(jobs []*jobState, stat stats.StatsReceiver, cs *clusterState,
-	requestors map[string][]*jobState, cfg SchedulerConfig) []*taskState {
+	requestors map[string][]*jobState) (startTasks []*taskState, stopTasks []*taskState) {
+
+	// Exit if there are no unscheduled tasks.
+	totalOutstandingTasks := 0
+	for _, j := range jobs {
+		totalOutstandingTasks += (len(j.Tasks) - j.TasksCompleted)
+	}
+
+	// Udate SoftMaxSchedulableTasks based on number of healthy nodes and the total number of tasks.
+	// Setting the max to num healthy nodes means that each job can be fully scheduled.
+	// (This gets used in config.GetNodeScaleFactor())
+	var cfg OrigSchedulingAlgConfig
+	if oa.Config != nil {
+		cfg = *oa.Config
+		if cfg.SoftMaxSchedulableTasks == 0 {
+			cfg.SoftMaxSchedulableTasks = DefaultSoftMaxSchedulableTasks
```

Comment:
does this actually modify oa.Config? or just the dereferenced value cfg

Reply:
removed lines 41/42 didn't need to add them